test/lib/analyzer/gomodules.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { DepGraph } from "@snyk/dep-graph";
+import { scan } from "../../../lib";
+import { DEP_GRAPH_TYPE } from "../../../lib/go-parser";
+import { getFixture } from "../../util";
+
+describe("goModulesToScannedProjects function", () => {
+  it("should return expected scannedProject[] result", async () => {
+    // Arrange
+    const fixturePath = getFixture("docker-archives/docker-save/yq.tar");
+    const imageNameAndTag = `docker-archive:${fixturePath}`;
+
+    // Act
+    const pluginResult = await scan({
+      path: imageNameAndTag,
+      "app-vulns": true,
+    });
```

Comment:
This looks like a full system tests which we already do in `test/system/application-scans/gomodules.spec.ts`, do you think there is any need for this one too?